src/protocols/bdx/AsyncTransferFacilitator.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "AsyncTransferFacilitator.h"
+
+#include <system/SystemClock.h>
+
+namespace chip {
+namespace bdx {
+
+AsyncTransferFacilitator::~AsyncTransferFacilitator()
+{
+    Reset();
+}
+
+void AsyncTransferFacilitator::Reset()
+{
+    mExchange.Release();
+    mTransfer.Reset();
+}
+
+CHIP_ERROR AsyncTransferFacilitator::SendMessage(TransferSession::OutputEvent & event)
+{
+
+    VerifyOrReturnError(mExchange, CHIP_ERROR_INCORRECT_STATE);
+
```

Comment:
Shouldn't we check the event type here?

Or better yet, only pass through the msgTypeData and MsgData and then you don't have to worry about this not being a message-send event....

Reply:
We do check if its a msgtosend then call this api. yes passing msgTypeData and MsgData should work.

Reply:
done.